### PR TITLE
fix :Upload count or number of contribution does not get updated when media  is successful uploaded

### LINKED ISSUE #1853: No of Uploads not updated
**Summary:** 
No of uploads doesn't increase after uploading the image.

**Steps to reproduce:** 
You can upload a new image and no of uploads count doesn't increase from the previous value.

**Device and Android version:** 

Tested On Redmi Note 4 (Nougat)

 **Commons app version:** 
 prodDebug.
 
**Screen-shots:** 
screenshot_2018-08-24-20-36-48-482_fr free nrw commons debug

**Would you like to work on the issue?**
 Yes, I would like to work on this issue.

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
Description (required)

Fixes #1853

What changes did you make and why?
* Add the function `upDateUploadCount ()` 
* Add javadocs

Tests performed (required)

Tested {build variant, e.g. ProdDebug,betaDebug} with API level {25,29}.

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul please review this one

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 1]
```diff
@@ -617,5 +621,18 @@ public void backButtonClicked() {
     public MediaDetailPagerFragment getMediaDetailPagerFragment() {
         return mediaDetailPagerFragment;
     }
+
+    /**
+     * this function updated the number of contributions
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:626]: updated -> updates

File: app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt [hunk 2]
```diff
@@ -141,6 +142,7 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
     }
 
     override suspend fun doWork(): Result {
+        var countUpload=0
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt:145]: Spaces before and after `=`

File: app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt [hunk 3]
```diff
@@ -180,12 +182,16 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
                         if (contribution.state == Contribution.STATE_QUEUED) {
                             contribution.state = Contribution.STATE_QUEUED_LIMITED_CONNECTION_MODE
                             contributionDao.save(contribution)
+                        } else {
+                            //To Nothing
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt:186]: Are these two lines really needed?

File: app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt [hunk 4]
```diff
@@ -180,12 +182,16 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
...
                         }
                     } else {
                         contribution.transferred = 0
                         contribution.state = Contribution.STATE_IN_PROGRESS
                         contributionDao.save(contribution)
                         uploadContribution(contribution = contribution)
+                        setProgressAsync(Data.Builder().putInt("progress", countUpload).build())
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt:193]: Please add a comment explaining what this line does.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt (hunk 3 above)
Comment (Prince-kushwaha) [reply app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt]: Yes.Else statement fix :'if' must have both main and 'else' branches if used as an expression

File: app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt (hunk 4 above)
Comment (Prince-kushwaha) [reply app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt:194]: This function updates the liveData when media is uploaded

**Comment by Prince-kushwaha** [comment]
@neslihanturan @madhurgupta10 up for review

**Review by nicolas-raoul — COMMENTED** [review]
I just tested this PR.
At the beginning I had 2773 contributions. I uploaded a set of 2 pictures, still 2773. Then I uploaded a set of 5 pictures, still 2773.
The strangest is that even after restarting the app it still says 2773, so I guess I am hitting a different bug?
Screenshot_20210605-112830_Commons
Screenshot_20210605-150319_Commons

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul it work prefect in my device on Redmi Note 4(API:25)

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul uninstalled the previous common   app then test it

**Comment by nicolas-raoul** [comment]
Would you mind fixing the conflict?

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul fix the conficts

**Review by nicolas-raoul — APPROVED** [review]
It works great, thanks!

**Pull request merged**